requirements.txt

```diff
@@ -5,4 +5,6 @@ inflection==0.5.1
 fluent-logger==0.10.0
 toml==0.10.2
 click==8.1.3
-semver==2.13.0
\ No newline at end of file
+semver==2.13.0
+langchain~=0.0.76
+transformers~=4.26.0
```

Comment:
Adding `langchain` and `transformers` to our requirements.txt makes me slightly nervous -- we're so dependency lean right now, but `langchain` alone is going to pull in:

* tensorflow
* pinecone
* wolfram
* numpy
* elasticsearch
* redis
* etc etc (full list copied below)

Which our plugin system works really hard to make unnecessary as things that get pulled in.

With apologies that this feels like a "thrash-inducing" PR question:

- Are there huge labor / UX downsides to considering the route where there's a separate PyPi module `steamship_langchain` that makes importing all these dependencies optional?


```
python = ">=3.8.1,<4.0"
pydantic = "^1"
SQLAlchemy = "^1"
requests = "^2"
PyYAML = "^6"
numpy = "^1"
faiss-cpu = {version = "^1", optional = true}
wikipedia = {version = "^1", optional = true}
elasticsearch = {version = "^8", optional = true}
redis = {version = "^4", optional = true}
manifest-ml = {version = "^0.0.1", optional = true}
spacy = {version = "^3", optional = true}
nltk = {version = "^3", optional = true}
transformers = {version = "^4", optional = true}
beautifulsoup4 = {version = "^4", optional = true}
torch = {version = "^1", optional = true}
jinja2 = {version = "^3", optional = true}
tiktoken = {version = "^0", optional = true, python="^3.9"}
pinecone-client = {version = "^2", optional = true}
weaviate-client = {version = "^3", optional = true}
google-api-python-client = {version = "2.70.0", optional = true}
wolframalpha = {version = "5.0.0", optional = true}
qdrant-client = {version = "^0.11.7", optional = true}
dataclasses-json = "^0.5.7"
tensorflow-text = {version = "^2.11.0", optional = true, python = "^3.10, <3.12"}
tenacity = "^8.1.0"
```

Reply:
I was worried about this as well.  I don't think it is a UX issue, really(though curious for other thoughts).  I don't know anything about the complexity of separating PyPi modules, but I'm happy to pursue that.